tests/test_routes.py

```diff
@@ -713,25 +715,37 @@ class TestExceptionHandler(RouteTest):
     def test_exception_handling(self):
         # The exception handler deals with most exceptions by running them
         # through ErrorHandler.handle()
-        value_error = ValueError()
+        assert isinstance(error_handler_object, ErrorHandler)
+
+        # Temporarily replace the ErrorHandler used by the
+        # exception_handler function -- this is what we imported as
+        # error_handler_object.
+        class MockErrorHandler(object):
+            def handle(self, exception):
+                self.handled = exception
+                return Response("handled it", 500)
+        routes.h = MockErrorHandler()
+
+        # Simulate a request that causes an unhandled exception.
         with self.app.test_request_context():
+            value_error = ValueError()
             result = exception_handler(value_error)
 
-            # This generally turns them into Internal Server Errors.
-            expect = error_handler_object.handle(value_error)
-            eq_(expect.data, result.data)
-            eq_(500, result.status_code)
-            eq_(expect.status_code, result.status_code)
+            # The exception was passed into MockErrorHandler.handle.
+            eq_(value_error, routes.h.handled)
 
-        # At this point the database session has been ruined, so
-        # we can't do any more checks in this test.
+            # The Response is created was passed along.
+            eq_("handled it", result.data)
+            eq_(500, result.status_code)
 
-    def test_exception_handling_http_exception(self):
```

Comment:
So this is removed because there are no http exceptions?

Reply:
No code is removed. I tried to split a test into two, which didn't solve the problem, so I merged them back to reduce the overall size of the diff.

Reply:
I was looking at the latest commit and on the entire PR before.